qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -259,32 +286,36 @@ testlog "TEST: cloned images"
 testlog " - default"
 parent_image=test_parent
 parent_snap=snap
-create_image ${CLUSTER2} ${PARENT_POOL} ${parent_image}
+create_image_and_enable_mirror ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${RBD_MIRROR_MODE}
 write_image ${CLUSTER2} ${PARENT_POOL} ${parent_image} 100
 create_snapshot ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap}
 protect_snapshot ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap}
 
 clone_image=test_clone
 clone_image ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} ${clone_image}
 write_image ${CLUSTER2} ${POOL} ${clone_image} 100
-
-enable_mirror ${CLUSTER2} ${PARENT_POOL} ${parent_image} journal
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  enable_mirror ${CLUSTER2} ${POOL} ${clone_image} snapshot
+else
+  enable_mirror ${CLUSTER2} ${PARENT_POOL} ${parent_image} journal
+fi
 wait_for_image_replay_started ${CLUSTER1} ${PARENT_POOL} ${parent_image}
 wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${PARENT_POOL} ${parent_image}
-wait_for_status_in_pool_dir ${CLUSTER1} ${PARENT_POOL} ${parent_image} 'up+replaying' 'primary_position'
+wait_for_replaying_status_in_pool_dir ${CLUSTER1} ${PARENT_POOL} ${parent_image}
 compare_images ${PARENT_POOL} ${parent_image}
 
 wait_for_image_replay_started ${CLUSTER1} ${POOL} ${clone_image}
 wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${clone_image}
-wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${clone_image} 'up+replaying' 'primary_position'
+wait_for_replaying_status_in_pool_dir ${CLUSTER1} ${POOL} ${clone_image}
 compare_images ${POOL} ${clone_image}
 remove_image_retry ${CLUSTER2} ${POOL} ${clone_image}
 
 testlog " - clone v1"
-clone_image ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} ${clone_image}1
-
-clone_image ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} \
-            ${clone_image}_v1 --rbd-default-clone-format 1
+clone_image_and_enable_mirror ${CLUSTER1} ${PARENT_POOL} \
+    ${parent_image} ${parent_snap} ${POOL} ${clone_image}1
```

Comment:
For consistency, let's pass `RBD_MIRROR_MODE` here too.

Reply:
Unresolving as this was missed.

Reply:
> Unresolving as this was missed.

Separately, it would be nice to clean up `RBD_MIRROR_MODE` vs `MIRROR_MODE_IMAGE` -- I'm pretty sure they are always the same.  But that can be done in a separate PR once the dust settles ;)

Reply:
@idryomov do you mean to change the above to:
```
 testlog " - clone v1"
 clone_image_and_enable_mirror ${CLUSTER1} ${PARENT_POOL} \
-    ${parent_image} ${parent_snap} ${POOL} ${clone_image}1
+    ${parent_image} ${parent_snap} ${POOL} ${clone_image}1 \
+    ${RBD_MIRROR_MODE} --rbd-default-clone-format 1
```
or
```
 testlog " - clone v1"
 clone_image_and_enable_mirror ${CLUSTER1} ${PARENT_POOL} \
-    ${parent_image} ${parent_snap} ${POOL} ${clone_image}1
+    ${parent_image} ${parent_snap} ${POOL} ${clone_image}1 \
+    ${RBD_MIRROR_MODE}
```

Reply:
The latter -- just tack on `RBD_MIRROR_MODE`.

Reply:
> Separately, it would be nice to clean up `RBD_MIRROR_MODE` vs `MIRROR_IMAGE_MODE` -- I'm pretty sure they are always the same.

https://tracker.ceph.com/issues/63431